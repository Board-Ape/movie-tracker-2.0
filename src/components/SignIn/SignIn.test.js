import React from 'react';
import { 
  SignIn,
  mapStateToProps,
  mapDispatchToProps } from './SignIn';
import { shallow } from 'enzyme';

describe('signin component tests', () => {
  let mockProps;
  let signInWrapper;
  beforeEach(() => {
    mockProps = {
      userObject: {
        error: false, signedIn: false, userData: { email: "123", id: 2, name: "123" }},
      signInToApp: jest.fn(),
      signUpForApp: jest.fn()
    };

    signInWrapper = shallow(<SignIn {...mockProps} />);
  });

  it('should render without crashing', () => {
    expect(signInWrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(signInWrapper).toMatchSnapshot();
  });

  it('should render the correct default elements', () => {
    const expectedFormLength = 1;
    const expectedInputLength = 4;

    expect(signInWrapper.find('form').length).toEqual(expectedFormLength);
    expect(signInWrapper.find('input').length).toEqual(expectedInputLength);
  });

  it('should focus an input when the inputFocus method is called', () => {
    const expectedPreFocusLength = 0;
    const expectedFocusLength = 1;

    expect(signInWrapper.find('.test-focus').length).toEqual(expectedPreFocusLength);

    signInWrapper.instance().inputFocus('username', 'signin-input test-focus');
    signInWrapper.update();

    expect(signInWrapper.find('.test-focus').length).toEqual(expectedFocusLength);
  });

  it('should toggle the signup type to true or false on changeSignInType call', () => {
    const expectedPreToggle = false;
    const expectedPostToggle = true;

    expect(signInWrapper.state('signup')).toEqual(expectedPreToggle);

    signInWrapper.instance().changeSignInType();
    signInWrapper.update();

    expect(signInWrapper.state('signup')).toEqual(expectedPostToggle);
  });

  it('should change the password input type based upon the event on the checkbox', () => {
    const expectedPreShowPass = 'password';
    const expectedPostShowPass = 'text';

    const mockEvent = { target: { checked: true }};

    expect(signInWrapper.state('passwordDisplay')).toEqual(expectedPreShowPass);

    signInWrapper.instance().handleShowPass(mockEvent);
    signInWrapper.update();

    expect(signInWrapper.state('passwordDisplay')).toEqual(expectedPostShowPass);
  });
});



describe('mapStateToProps tests', () => {
  it('should map in the user object from store', () => {
    const mockStore = {
      SignIn: {
        error: false, 
        signedIn: false, 
        userData: { email: "123", id: 2, name: "123" }}
    };

    const result = mapStateToProps(mockStore);
    expect(result.userObject).toEqual(mockStore.SignIn);
  });
});

describe('mapDispatchToProps tests', () => {
  it('should map dispatch to props for signin functions', () => {
    const mockDispatch = jest.fn();
    const result = mapDispatchToProps(mockDispatch);

    result.signInToApp('user', 'pass');
    
    expect(mockDispatch).toHaveBeenCalled();
  });
  
  it('should map dispatch to props for signup functions', () => {
    const mockDispatch = jest.fn();
    const result = mapDispatchToProps(mockDispatch);

    result.signUpForApp('name', 'user', 'pass');
    
    expect(mockDispatch).toHaveBeenCalled();
  });
});
