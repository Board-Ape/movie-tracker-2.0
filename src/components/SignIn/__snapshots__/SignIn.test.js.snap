// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signin component tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    signInToApp={[Function]}
    signUpForApp={[Function]}
    userObject={
        Object {
            "error": false,
            "signedIn": false,
            "userData": Object {
              "email": "123",
              "id": 2,
              "name": "123",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="in"
        onSubmit={[Function]}
>
        <Link
                className="sign-in-close"
                replace={false}
                to="/"
        >
                Close
        </Link>
        <input
                className="signin-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Username (email)"
                type="text"
                value=""
        />
        <input
                className="signin-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Password (case sensitive)"
                type="password"
                value=""
        />
        <div
                className="password-show-hide"
        >
                <div
                        className="type-toggle in"
                >
                        <button
                                className="in"
                                onClick={[Function]}
                                tabIndex="-1"
                        >
                                Sign Up Instead
                        </button>
                </div>
                <label
                        htmlFor="password-option"
                >
                        Show Password?
                </label>
                <input
                        id="password-option"
                        name="passOption"
                        onChange={[Function]}
                        tabIndex="-1"
                        type="checkbox"
                        value="text"
                />
        </div>
        <input
                type="submit"
                value="Sign In!"
        />
</form>,
      "className": "SignIn",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="sign-in-close"
            replace={false}
            to="/"
>
            Close
</Link>,
          null,
          null,
          null,
          <input
            className="signin-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Username (email)"
            type="text"
            value=""
/>,
          <input
            className="signin-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password (case sensitive)"
            type="password"
            value=""
/>,
          <div
            className="password-show-hide"
>
            <div
                        className="type-toggle in"
            >
                        <button
                                    className="in"
                                    onClick={[Function]}
                                    tabIndex="-1"
                        >
                                    Sign Up Instead
                        </button>
            </div>
            <label
                        htmlFor="password-option"
            >
                        Show Password?
            </label>
            <input
                        id="password-option"
                        name="passOption"
                        onChange={[Function]}
                        tabIndex="-1"
                        type="checkbox"
                        value="text"
            />
</div>,
          <input
            type="submit"
            value="Sign In!"
/>,
        ],
        "className": "in",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Close",
            "className": "sign-in-close",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Close",
          "type": [Function],
        },
        null,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signin-input",
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "placeholder": "Username (email)",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signin-input",
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "placeholder": "Password (case sensitive)",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="type-toggle in"
>
                <button
                                className="in"
                                onClick={[Function]}
                                tabIndex="-1"
                >
                                Sign Up Instead
                </button>
</div>,
              <label
                htmlFor="password-option"
>
                Show Password?
</label>,
              <input
                id="password-option"
                name="passOption"
                onChange={[Function]}
                tabIndex="-1"
                type="checkbox"
                value="text"
/>,
            ],
            "className": "password-show-hide",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="in"
                  onClick={[Function]}
                  tabIndex="-1"
>
                  Sign Up Instead
</button>,
                "className": "type-toggle in",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up Instead",
                  "className": "in",
                  "onClick": [Function],
                  "tabIndex": "-1",
                },
                "ref": null,
                "rendered": "Sign Up Instead",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Show Password?",
                "htmlFor": "password-option",
              },
              "ref": null,
              "rendered": "Show Password?",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password-option",
                "name": "passOption",
                "onChange": [Function],
                "tabIndex": "-1",
                "type": "checkbox",
                "value": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "Sign In!",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="in"
          onSubmit={[Function]}
>
          <Link
                    className="sign-in-close"
                    replace={false}
                    to="/"
          >
                    Close
          </Link>
          <input
                    className="signin-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Username (email)"
                    type="text"
                    value=""
          />
          <input
                    className="signin-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Password (case sensitive)"
                    type="password"
                    value=""
          />
          <div
                    className="password-show-hide"
          >
                    <div
                              className="type-toggle in"
                    >
                              <button
                                        className="in"
                                        onClick={[Function]}
                                        tabIndex="-1"
                              >
                                        Sign Up Instead
                              </button>
                    </div>
                    <label
                              htmlFor="password-option"
                    >
                              Show Password?
                    </label>
                    <input
                              id="password-option"
                              name="passOption"
                              onChange={[Function]}
                              tabIndex="-1"
                              type="checkbox"
                              value="text"
                    />
          </div>
          <input
                    type="submit"
                    value="Sign In!"
          />
</form>,
        "className": "SignIn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="sign-in-close"
              replace={false}
              to="/"
>
              Close
</Link>,
            null,
            null,
            null,
            <input
              className="signin-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Username (email)"
              type="text"
              value=""
/>,
            <input
              className="signin-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Password (case sensitive)"
              type="password"
              value=""
/>,
            <div
              className="password-show-hide"
>
              <div
                            className="type-toggle in"
              >
                            <button
                                          className="in"
                                          onClick={[Function]}
                                          tabIndex="-1"
                            >
                                          Sign Up Instead
                            </button>
              </div>
              <label
                            htmlFor="password-option"
              >
                            Show Password?
              </label>
              <input
                            id="password-option"
                            name="passOption"
                            onChange={[Function]}
                            tabIndex="-1"
                            type="checkbox"
                            value="text"
              />
</div>,
            <input
              type="submit"
              value="Sign In!"
/>,
          ],
          "className": "in",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "className": "sign-in-close",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
          null,
          null,
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signin-input",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "placeholder": "Username (email)",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signin-input",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "placeholder": "Password (case sensitive)",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="type-toggle in"
>
                  <button
                                    className="in"
                                    onClick={[Function]}
                                    tabIndex="-1"
                  >
                                    Sign Up Instead
                  </button>
</div>,
                <label
                  htmlFor="password-option"
>
                  Show Password?
</label>,
                <input
                  id="password-option"
                  name="passOption"
                  onChange={[Function]}
                  tabIndex="-1"
                  type="checkbox"
                  value="text"
/>,
              ],
              "className": "password-show-hide",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="in"
                    onClick={[Function]}
                    tabIndex="-1"
>
                    Sign Up Instead
</button>,
                  "className": "type-toggle in",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Up Instead",
                    "className": "in",
                    "onClick": [Function],
                    "tabIndex": "-1",
                  },
                  "ref": null,
                  "rendered": "Sign Up Instead",
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Show Password?",
                  "htmlFor": "password-option",
                },
                "ref": null,
                "rendered": "Show Password?",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "password-option",
                  "name": "passOption",
                  "onChange": [Function],
                  "tabIndex": "-1",
                  "type": "checkbox",
                  "value": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Sign In!",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
